### PR TITLE
Extract Accounts into an interface for easier testing

### PR DESCRIPTION
**Pull request by m4ksio**
That's for upcoming PR with small fix and tests

### CODE REVIEW

File: fvm/state/accounts.go [hunk 1]
```diff
@@ -32,17 +32,38 @@ func keyPublicKey(index uint64) string {
...
+	CheckAccountNotFrozen(address flow.Address) error
+	GetStorageUsed(address flow.Address) (uint64, error)
+	SetValue(address flow.Address, key string, value []byte) error
+	AllocateStorageIndex(address flow.Address) (uint64, error)
+	SetAccountFrozen(address flow.Address, frozen bool) error
+}
+
+var _ Accounts = &AccountsState{}
+
+type AccountsState struct {
```
Comment (ramtinms) [inline fvm/state/accounts.go:56]: Maybe another name like StatefulAccounts

File: fvm/state/accounts.go [hunk 2]
```diff
@@ -32,17 +32,38 @@ func keyPublicKey(index uint64) string {
 	return fmt.Sprintf("public_key_%d", index)
 }
 
-type Accounts struct {
+type Accounts interface {
```
Comment (ramtinms) [inline fvm/state/accounts.go:35]: We actually were planing last year to break this into smaller interfaces like AccountValues, AccountKeys, AccountContracts, AccountMeta, ... and Accounts to be an interface of all other interfaces.

**Review by ramtinms — APPROVED** [review]

File: cmd/util/ledger/migrations/storage_v5.go [hunk 3]
```diff
@@ -160,7 +160,7 @@ func (m *StorageFormatV5Migration) Migrate(payloads []ledger.Payload) ([]ledger.
 	return migratedPayloads, nil
 }
 
-func (m StorageFormatV5Migration) getContractsOnlyAccounts(payloads []ledger.Payload) *state.Accounts {
+func (m StorageFormatV5Migration) getContractsOnlyAccounts(payloads []ledger.Payload) *state.AccountsState {
```
Comment (janezpodhostnik) [inline cmd/util/ledger/migrations/storage_v5.go:163]: This return type could be `state.Accounts` which would mean `AccountsState` wouldn't need to be exported from the fvm.

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/state/accounts.go (hunk 1 above)
Comment (m4ksio) [reply fvm/state/accounts.go:56]: Yep, I like it

**Pull request merged**